docs/DeveloperGuide.adoc

```diff
@@ -153,98 +173,236 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+** Pros: Easy to organize and the UI is clean.
+** Cons: Less interaction, and clicking is required. While allowing user to use command to switch among different tabs is under construction.
+* **Alternative 2:** Use a simple `panel` to hold the chart, and allow user to use command to toggle the type of charts.
+** Pros: More flexible and more interaction possibilities.
+** Cons: The cost overweight the benefit.
+
+==== Future possibility
+===== Question: should we write this in DG?
+// end::taskSummary[]
+
+
 
-===== Aspect: How undo & redo executes
+//tag:calendar[]
+=== Calendar feature
 
-* **Alternative 1 (current choice):** Saves the entire address book.
-** Pros: Easy to implement.
-** Cons: May have performance issues in terms of memory usage.
-* **Alternative 2:** Individual command knows how to undo/redo by itself.
-** Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-** Cons: We must ensure that the implementation of each individual command are correct.
+==== Implementation
+The calendar feature, as the name suggests, displays a calendar fo users.
 
-===== Aspect: Data structure to support the undo/redo commands
+The feature is implemented by overriding the task list display panel on the main window with calendar box. The calendar currently supports following features:
 
-* **Alternative 1 (current choice):** Use a list to store the history of address book states.
-** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedAddressBook`.
-* **Alternative 2:** Use `HistoryManager` for undo/redo
-** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
-** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
-// end::undoredo[]
+. Displays the name of all tasks on that day in the calendar grid.
+. Access calendars in previous or next month using the buttons on top.
+. Displays more information about tasks for the day on the due soon panel after clicking on a grid. (Not done)
+. Fast forward to a specific year/month using CLI. (Not done)
+
+==== Example usage scenario
+Clicking on the show calendar menu item under the calendar menu will enable this feature. Upon requesting for the calendar to be shown:
+
+. The calendar box object will be constructed
+. The calendar loops through every day of the month while searching for tasks due on that day
+. The task name is added to the calendar grid
+
+Activity diagram:
+
+image::CalendarActivityDiagram.png[width = "400", length = "600"]
```

Comment:
I believe the command box for (day < Month.length) is not compliant with the activity diagram we were taught? Also, the if else condition should have square bracket.